packages/superset-ui-core/test/query/buildQueryObject.test.ts

```diff
@@ -58,6 +58,41 @@ describe('buildQueryObject', () => {
     expect(query.metrics).toEqual(['sum__num', 'avg__num']);
   });
 
+  it('should merge original and append filters', () => {
+    query = buildQueryObject({
+      datasource: '5__table',
+      granularity_sqla: 'ds',
+      viz_type: 'table',
+      filters: [{ col: 'abc', op: '==', val: 'qwerty' }],
+      adhoc_filters: [
+        {
+          expressionType: 'SIMPLE',
+          clause: 'WHERE',
+          subject: 'foo',
+          operator: '!=',
+          comparator: 'bar',
+        },
+      ],
+      where: 'a = b',
+      extra_form_data: {
+        append_form_data: {
+          adhoc_filters: [
+            {
+              expressionType: 'SQL',
+              clause: 'WHERE',
+              sqlExpression: '(1 = 1)',
+            },
+          ],
+        },
```

Comment:
Question about `adhoc_filters` and `filters`. Is my understanding correct that `filters` are only used for native filters and can only be set from `append_form_data`, since all filters defined in chart `formData` are basically `adhoc_filters`? 

I'm a little confused why do we need to allow appending and overriding both `filters` and `adhoc_filters`. The overrides are only from a Dashboard's native/cross filters, which would be only in `QueryObjectFilterClause` or `QueryFormExtraFilter` format.

It seems to me the original design of an `adhoc_filters` vs `extra_filters` was cleaner. 

Not sure if this is the direction of future refactoring, but we probably shouldn't add extra stuff that are not configurable by chart controls to `formData`. This `extra_form_data ` could be moved to the extra options of `buildQueryObject` and `buildQuery`. E.g.

```ts
buildQueryObject(formData, {
  queryFields: ...         // configured in chart metatada,
  extraFormData: {
    append: {
      adhoc_filter: [...],
    },
    override: {
    }
  }                         // collected from dashboard state
});
```

Reply:
AFAIK `filters` are just the simple filters that preceeded `adhoc_filters` (before my time). I'm not sure why we support both, as `adhoc_filters` are a superset of `filters`, but my assumption is it's a leftover from old times. I've been using the `filters` property when adding filters that are simple, i.e. don't require custom SQL, but we could equally well ditch `filters` and just make it possible append to `adhoc_filters`.

Reply:
We should probably:

1. Rename the Python attribute [`query_object.filter`](https://github.com/apache/superset/blob/694ae6f90e76b864d5d721c4223d36e01f7fd4cf/superset/common/query_object.py#L178) to `query_object.filters`
2. Rename `adhoc_filters` on the client side to `filters` and have it support both simple filter and adhoc filters.
3. Move the filter format consolidation logic to the server side. Either in QueryObject itself or when actually building the SQL queries.